<a href="https://colab.research.google.com/github/barbycardozo81/InicioPython/blob/main/panda.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd
# Cargamos uno de los archivos dentro del dataset
df = pd.read_csv('./data/GlobalLandTemperaturesByCountry.csv')
# Mostrar las primeras 6 filas
print(df.head(6))
# Información del dataset
print("\nInformación del archivo:")
df.info()
# Cantidad total de datos (filas y columnas)
print("\nForma del dataset (filas, columnas):", df.shape)

FileNotFoundError: [Errno 2] No such file or directory: './data/GlobalLandTemperaturesByCountry.csv'

In [4]:
import os

# Crear el directorio 'data' si no existe
if not os.path.exists('./data'):
    os.makedirs('./data')

# Descargar el archivo CSV desde una fuente pública accesible
# URL corregida: Este enlace apunta a un repositorio de GitHub que contiene el archivo.
!wget -O ./data/GlobalLandTemperaturesByCountry.csv https://raw.githubusercontent.com/Manish-Bhutani/Climate-Change-Earth-Surface-Temperature-Data/master/GlobalLandTemperaturesByCountry.csv

--2026-05-05 08:13:08--  https://raw.githubusercontent.com/Manish-Bhutani/Climate-Change-Earth-Surface-Temperature-Data/master/GlobalLandTemperaturesByCountry.csv
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
2026-05-05 08:13:08 ERROR 404: Not Found.



In [6]:
import pandas as pd

# Crear un DataFrame de ejemplo con datos de estudiantes
data = {
    'student_id': [1, 2, 3, 4, 5],
    'name': ['Ana', 'Juan', 'María', 'Pedro', 'Laura'],
    'age': [15, 17, 16, 18, 15],
    'nota': [7, 5, 8, 9, 6]
}
sample_df = pd.DataFrame(data)

# Guardar el DataFrame en un archivo Excel en /content/
sample_df.to_excel('/content/Maths.xlsx', index=False)

print("Archivo 'Maths.xlsx' de ejemplo creado en /content/.")

Archivo 'Maths.xlsx' de ejemplo creado en /content/.


In [5]:
import pandas as pd
# Cargar archivo excel (leer datos desde un archivo)
df = pd.read_excel("/content/Maths.xlsx")
# Mostrar las primeras filas del DataFrame
# head() sirve para ver rápidamente cómo son los datos
print("Primeras filas:")
print(df.head())
# Filtrar alumnos aprobados (nota >= 6)
edad = df[df["age"] >= 16]
print("\nAlumnos mayores de 16 años:")
print(edad)

FileNotFoundError: [Errno 2] No such file or directory: '/content/Maths.xlsx'

El archivo `GlobalLandTemperaturesByCountry.csv` no se pudo descargar en los intentos anteriores (código de error 404 Not Found), lo que significa que la URL proporcionada no contenía el archivo. Sigo buscando una fuente pública confiable para ese archivo. Si tienes una URL diferente o el archivo localmente, por favor, proporciónamela para continuar.